<!-- a collection of all mouseover info popups -->

<div class="modal fade" id="job_name_info" tabindex="-1" role="dialog" aria-labelledby="job_name_info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">You can give your job a name to better distinguish between multiple runs.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="InputSequence_Info" tabindex="-1" role="dialog" aria-labelledby="InputSequence_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">The input sequence file in fasta format.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="DBselectorInfo" tabindex="-1" role="dialog"
      aria-labelledby="DBSelectorInfo">
     <div class="modal-dialog" role="document">
         <div class="modal-content">
             <div class="modal-body">
                 <font color="#0B0719">
                     <p align="justify"> Motif database used for annotating trained models.</p>
                 </font>
             </div>
         </div>
     </div>
 </div>

<div class="modal fade" id="Background_Sequences_Info" tabindex="-1" role="dialog"
     aria-labelledby="BackgroundSequence_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">A fasta file with sequences that represent the genomic background of the input
                        sequences. If not provided, the background is estimated from the input sequences.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="MotifInitialization_Info" tabindex="-1" role="dialog"
     aria-labelledby="MotifInitialization_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Please select if you want PEnGmotif be in charge of the Motif Initialization
                        or provide a custom Initialization File. The initialization file can be a list of binding sites, a PWM or
                        BaMM file.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="MotifInitFile_Info" tabindex="-1" role="dialog" aria-labelledby="MotifInitFile_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">If you have a list of binding sites, a PWM or a BaMM model which you want to refine insert it
                        here. Alternatively you can leave this empty and PEnGmotif will seed your input
                        sequence set.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="MotifInitFileFormat_Info" tabindex="-1" role="dialog"
     aria-labelledby="MotifInitFileFormat_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Please select the file format of your provided motif Initialization. This
                        option is only necessary when providing a custom MotifInitialization.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="ModelOrder_Info" tabindex="-1" role="dialog" aria-labelledby="ModelOrder_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">
                      The highest order of the BaMM. A model of order 0 corresponds to a PWM model.
                    </p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="ReverseComplement_Info" tabindex="-1" role="dialog"
     aria-labelledby="ReverseComplement_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> When checked, motifs can be both on the plus and minus strand.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="Extend_Info" tabindex="-1" role="dialog" aria-labelledby="Extend_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Extends the motif by adding extra positions to the left
                        and right of the initializer motif. If the initializer only consists of the informative core motif, enlarging the motif allows to also learn the flanking regions.
                    </p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="FDR_Info" tabindex="-1" role="dialog" aria-labelledby="FDR_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Evaluate the performance of the motif on the input set. Required for plotting performance plots.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="mFold_Info" tabindex="-1" role="dialog" aria-labelledby="mFold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> The number of negative sequences as multiple of positive sequences used
                        during FDR calculation.
                        The default is 5.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="cvFold_Info" tabindex="-1" role="dialog" aria-labelledby="cvFold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Number of cross-validation folds during FDR calculation.
                        The default is 5.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="BackgroundOrder_Info" tabindex="-1" role="dialog"
     aria-labelledby="BackgroundOrder_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">
                      The order for the background model. We recommend a background order of 2 for a realistic model of the genomic input.
                      For very short motifs a background order of 1 or even 0 may be required to find the motif.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="FDRSamplingOrder_Info" tabindex="-1" role="dialog"
     aria-labelledby="FDRSamplingOrder_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Order of kmer for sampling negative sets during FDR calculation.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EMq_Info" tabindex="-1" role="dialog" aria-labelledby="EMq_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Prior probability for a positive sequence to contain a motif. The default is
                        0.9 which is the expected probability of a successfull ChIPseq experiment.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EMepsilon_Info" tabindex="-1" role="dialog" aria-labelledby="EMepsilon_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> The EM algorithm is deemed to be converged when the sum over the
                        absolute differences in BaMM probabilities from successive EM rounds
                        is smaller than epsilon. The default is 0.001</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EMmaxIter_Info" tabindex="-1" role="dialog" aria-labelledby="EMmaxIter_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Limit the number of EM iterations.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="MMcompare_Info" tabindex="-1" role="dialog" aria-labelledby="MMcompare_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">When checked, the inserted Motif InitFile will be compared to our BaMMmotif
                        Database to find similarities.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="db_match_p_value_cutoff_Info" tabindex="-1" role="dialog"
     aria-labelledby="db_match_p_value_cutoff_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">The p-value limit will be used to define a threshold for motif comparisons
                        between the inserted model and the database. </p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="score_Cutoff_Info" tabindex="-1" role="dialog"
     aria-labelledby="db_match_p_value_cutoff_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Only motif positions with an e-value smaller than this will be reported as binding positions. </p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="BaMMScan_Info" tabindex="-1" role="dialog"
     aria-labelledby="db_match_p_value_cutoff_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Scan the input sequences for motif occurences. Required for plotting motif localization.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="bgModel_File_Info" tabindex="-1" role="dialog"
     aria-labelledby="db_match_p_value_cutoff_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">Only needed when providing a BaMM model as initialization. Enter the background model file which belongs to the BaMM model provided as initialization.  </p>
                </font>
            </div>
        </div>
    </div>
</div>


<div class="modal fade" id="InputSequence_Info" tabindex="-1" role="dialog" aria-labelledby="InputSequence_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify">The Input Sequence File needs to be of Fasta Format.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="ReverseComplement_Info" tabindex="-1" role="dialog"
     aria-labelledby="RevverseComplement_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> When checked, the reverse complementary sequences will be used as well for motif search.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EM_Info" tabindex="-1" role="dialog" aria-labelledby="EM_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> When checked, the Expectation Maximization step will be
                        skipped.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EM_Max_Iterations_Info" tabindex="-1" role="dialog" aria-labelledby="EMmaxIter_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Maximum number of Expectation Maximization iterations.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="Pattern_Length_Info" tabindex="-1" role="dialog" aria-labelledby="Pattern_Length_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Length of patterns to be searched.</p>
                </font>
            </div>
        </div>
    </div>
</div>


<div class="modal fade" id="Z-Score_Threshold_Info" tabindex="-1" role="dialog" aria-labelledby="Z-Score_Threshold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Minimum z-score threshold for base patterns.</p>
                </font>
            </div>
        </div>
    </div>
</div>


<div class="modal fade" id="Count_Threshold_Info" tabindex="-1" role="dialog" aria-labelledby="Count_Threshold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Patterns with less than this amount of counts will not be considered.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="IUPAC_Optimization_Score_Info" tabindex="-1" role="dialog" aria-labelledby="IUPAC_Optimization_Score_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> The scoring function used for the optimnization to IUPAC patterns.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EM_Threshold_Info" tabindex="-1" role="dialog" aria-labelledby="EM_Threshold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> threshold for finishing the em optimization.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="Merging_Info" tabindex="-1" role="dialog" aria-labelledby="Merging_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Turns on the merging.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="Bit_Factor_Threshold_Info" tabindex="-1" role="dialog" aria-labelledby="Bit_Factor_Threshold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> bit factor threshold for merging IUPAC patterns.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="Use_Default_PWM_Info" tabindex="-1" role="dialog" aria-labelledby="Use_Default_PWM_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> use the default calculation of the ppm.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="PWM_Pseudo_Counts_Info" tabindex="-1" role="dialog" aria-labelledby="PWM_Pseudo_Counts_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> number of pseudo-counts for optimization.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="EM_Saturation_Threshold_Info" tabindex="-1" role="dialog" aria-labelledby="EM_Saturation_Threshold_Info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> saturation factor for em optimization.</p>
                </font>
            </div>
        </div>
    </div>
</div>

<div class="modal fade" id="max_optimized_seeds_info" tabindex="-1" role="dialog" aria-labelledby="max_optimized_seeds_info">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-body">
                <font color="#0B0719">
                    <p align="justify"> Maximum number of seeds that will be optimized.</p>
                </font>
            </div>
        </div>
    </div>
</div>
